File: base/src/main/java/org/jcodec/scale/RgbToBgr.java
package org.jcodec.scale;
import org.jcodec.api.NotSupportedException;
import org.jcodec.common.model.ColorSpace;
import org.jcodec.common.model.Picture;

import java.lang.IllegalArgumentException;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * @author The JCodec project
 *
 */
@Deprecated
public class RgbToBgr implements Transform {

    @Override
    public void transform(Picture src, Picture dst) {
        if (src.getColor() != ColorSpace.RGB && src.getColor() != ColorSpace.BGR
                || dst.getColor() != ColorSpace.RGB && dst.getColor() != ColorSpace.BGR) {
            throw new IllegalArgumentException(
                    "Expected RGB or BGR inputs, was: " + src.getColor() + ", " + dst.getColor());
        }
        if (src.getCrop() != null || dst.getCrop() != null)
            throw new NotSupportedException("Cropped images not supported");

        int[] dataSrc = src.getPlaneData(0);
        int[] dataDst = dst.getPlaneData(0);
        for (int i = 0; i < dataSrc.length; i += 3) {
            // src and dst can actually be the same array
            int tmp = dataSrc[i + 2];
            dataDst[i + 2] = dataSrc[i];
            dataDst[i] = tmp;
        }
    }

}

File: src/main/java/org/jcodec/scale/RgbToBgr.kt
package org.jcodec.scale

import org.jcodec.common.model.ColorSpace
import org.jcodec.common.model.Picture

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * @author The JCodec project
 */
class RgbToBgr : Transform {
    override fun transform(src: Picture, dst: Picture) {
        require(!(src.color != ColorSpace.RGB && src.color != ColorSpace.BGR
                || dst.color != ColorSpace.RGB && dst.color != ColorSpace.BGR)) { "Expected RGB or BGR inputs, was: " + src.color + ", " + dst.color }
        val dataSrc = src.getPlaneData(0)
        val dataDst = dst.getPlaneData(0)
        var i = 0
        while (i < dataSrc.size) {
            val tmp = dataSrc[i + 2]
            dataDst[i + 2] = dataSrc[i]
            dataDst[i] = tmp
            dataDst[i + 1] = dataSrc[i + 1]
            i += 3
        }
    }
}